dash/_callback.py

```diff
@@ -452,35 +467,49 @@ def add_context(*args, **kwargs):
             if NoUpdate.is_no_update(output_value):
                 raise PreventUpdate
 
-            if not multi:
-                output_value, output_spec = [output_value], [output_spec]
-                flat_output_values = output_value
-            else:
-                if isinstance(output_value, (list, tuple)):
-                    # For multi-output, allow top-level collection to be
-                    # list or tuple
-                    output_value = list(output_value)
-
-                # Flatten grouping and validate grouping structure
-                flat_output_values = flatten_grouping(output_value, output)
+            component_ids = collections.defaultdict(dict)
 
-            _validate.validate_multi_return(
-                output_spec, flat_output_values, callback_id
-            )
+            if has_output:
+                if not multi:
+                    output_value, output_spec = [output_value], [output_spec]
+                    flat_output_values = output_value
+                else:
+                    if isinstance(output_value, (list, tuple)):
+                        # For multi-output, allow top-level collection to be
+                        # list or tuple
+                        output_value = list(output_value)
+
+                    # Flatten grouping and validate grouping structure
+                    flat_output_values = flatten_grouping(output_value, output)
+
+                _validate.validate_multi_return(
+                    output_spec, flat_output_values, callback_id
+                )
 
-            component_ids = collections.defaultdict(dict)
-            has_update = False
-            for val, spec in zip(flat_output_values, output_spec):
-                if isinstance(val, NoUpdate):
-                    continue
-                for vali, speci in (
-                    zip(val, spec) if isinstance(spec, list) else [[val, spec]]
-                ):
-                    if not isinstance(vali, NoUpdate):
-                        has_update = True
-                        id_str = stringify_id(speci["id"])
-                        prop = clean_property_name(speci["property"])
-                        component_ids[id_str][prop] = vali
+                for val, spec in zip(flat_output_values, output_spec):
+                    if isinstance(val, NoUpdate):
+                        continue
+                    for vali, speci in (
+                        zip(val, spec) if isinstance(spec, list) else [[val, spec]]
+                    ):
+                        if not isinstance(vali, NoUpdate):
+                            has_update = True
+                            id_str = stringify_id(speci["id"])
+                            prop = clean_property_name(speci["property"])
+                            component_ids[id_str][prop] = vali
+            else:
+                if output_value is not None:
+                    raise InvalidCallbackReturnValue(
```

Comment:
This gives a weird error message: 
<img width="599" alt="Screen Shot 2024-04-30 at 5 47 28 PM" src="https://github.com/plotly/dash/assets/4672118/1527b801-4bbe-47c6-819a-2fc9cb26e124">

Not crucial to fix as part of this PR but would be nice if it's an easy fix.